megamek/src/megamek/common/MechSearchFilter.java

```diff
@@ -229,76 +345,141 @@ public static boolean isTechMatch(MechSummary mech, int nTechType) {
-            } else if (f.iWalk == 1) { // equal to
-                if (walk != mech.getWalkMp()) {
-                    return false;
-                }
-            } else if (f.iWalk == 2) { // not more than
-                if (mech.getWalkMp() > walk) {
-                    return false;
-                }
-            }
+    }
+
+    private static boolean isBetween(double value, String sStart, String sEnd) {
+        if (sStart.isEmpty() && sEnd.isEmpty()) {
+            return true;
         }
 
-        // Check jump criteria
-        int jump = -1;
-        try {
-            jump = Integer.parseInt(f.sJump);
-        } catch (Exception ignored) {
+        int iStart = toInt(sStart, Integer.MIN_VALUE);
+        int iEnd = toInt(sEnd, Integer.MAX_VALUE);
 
+        if ((value < iStart) || (value > iEnd)) {
+            return false;
         }
 
-        if (jump > -1) {
-            if (f.iJump == 0) { // at least
-                if (mech.getJumpMp() < jump) {
-                    return false;
-                }
-            } else if (f.iJump == 1) { // equal to
-                if (jump != mech.getJumpMp()) {
-                    return false;
-                }
-            } else if (f.iJump == 2) { // not more than
-                if (mech.getJumpMp() > jump) {
-                    return false;
-                }
+        return true;
+    }
+
+    private static boolean isMatch(int i, boolean b) {
+        if (i == 1) {
+            return b;
+        } else if (i == 2) {
+            return !b;
+        }
+
+        return true;
+    }
+
+    private static boolean checkA(List<String> list, String search, boolean b) {
+        if (list.isEmpty()) {
+            return true;
+        }
+
+        boolean match = false;
+
+        for (String s : list) {
+            if (search.contains(s) == b) {
+                match = true;
+                break;
             }
         }
 
-        if (f.checkInternalsType) {
-            if (f.internalsType != mech.getInternalsType()) {
+        if (!match) {
+            return false;
+        }
+
+        return match;
+    }
+
+    private static boolean checkB(List<String> list, String search, boolean b) {
+        for (String s : list) {
+            if (search.contains(s) == b) {
                 return false;
             }
         }
 
-        if (f.checkArmorType) {
-            if (!mech.getArmorType().contains(f.armorType)) {
+        return true;
+    }
+
+    private static boolean checkA(List<Integer> list, int search) {
+        if (list.isEmpty()) {
+            return true;
+        }
+
+        boolean match = false;
+
+        for (int s : list) {
+            if (search == s) {
+                match = true;
+                break;
+            }
+        }
+
+        if (!match) {
+            return false;
+        }
+
+        return match;
+    }
+
+    private static boolean checkB(List<Integer> list, int search) {
+        for (int s : list) {
```

Comment:
Isn't this the same as
return !list.contains(search)
?

Also, please try using descriptive names if possible. search could be item or searchItem. Also, what is checkB? It will be helpful at a later point.